kube-runtime/src/controller/mod.rs

```diff
@@ -22,38 +22,30 @@ use futures::{
 };
 use kube_client::api::{Api, DynamicObject, ListParams, Resource};
 use serde::de::DeserializeOwned;
-use snafu::{futures::TryStreamExt as SnafuTryStreamExt, Backtrace, ResultExt, Snafu};
 use std::{
     fmt::{Debug, Display},
     hash::Hash,
     sync::Arc,
     time::Duration,
 };
 use stream::BoxStream;
+use thiserror::Error;
 use tokio::{runtime::Handle, time::Instant};
 use tracing::{info_span, Instrument};
 
 mod future_hash_map;
 mod runner;
 
-#[derive(Snafu, Debug)]
+#[derive(Debug, Error)]
 pub enum Error<ReconcilerErr: std::error::Error + 'static, QueueErr: std::error::Error + 'static> {
-    ObjectNotFound {
-        obj_ref: ObjectRef<DynamicObject>,
-        backtrace: Backtrace,
-    },
-    ReconcilerFailed {
-        source: ReconcilerErr,
-        backtrace: Backtrace,
-    },
-    SchedulerDequeueFailed {
-        #[snafu(backtrace)]
-        source: scheduler::Error,
-    },
-    QueueError {
-        source: QueueErr,
-        backtrace: Backtrace,
-    },
+    #[error("ObjectNotFound")]
+    ObjectNotFound(ObjectRef<DynamicObject>),
+    #[error("ReconcilerFailed: {0}")]
+    ReconcilerFailed(#[source] ReconcilerErr),
+    #[error("SchedulerDequeueFailed: {0}")]
+    SchedulerDequeueFailed(#[source] scheduler::Error),
+    #[error("QueueError: {0}")]
+    QueueError(#[source] QueueErr),
```

Comment:
I think tuples are fine here, but we can do struct variants if that's better. 

Added `Display` to match `snafu`'s default (variant name, then inner error if there's a source).

Reply:
Imo this is fine. Tuples matches the approach we follow elsewhere.